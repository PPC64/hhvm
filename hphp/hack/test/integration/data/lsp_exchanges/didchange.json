[
{
  "jsonrpc": "2.0",
  "id": 1,
  "method": "initialize",
  "params":
    {
      "initializationOptions":{},
      "processId":null,
      "rootPath":"${root_path}",
      "capabilities":{}
    }
},
{
  "jsonrpc": "2.0",
  "method": "textDocument/didOpen",
  "params": {
     "textDocument": {
       "uri":"${php_file_uri}",
       "languageId": "hack",
       "version":1,
       "text":"${php_file}"
     }
   }
},
{
  "jsonrpc": "2.0",
  "method": "textDocument/didChange",
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
       },
       "contentChanges": [{
         "range": {
           "start": {"line":2, "character":9},
           "end": {"line":2, "character":12}
         },
         "text":"***"
       }]
   }
}
]
