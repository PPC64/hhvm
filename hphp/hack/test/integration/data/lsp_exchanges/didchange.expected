[
  {
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
      "capabilities": {
        "textDocumentSync": {
          "openClose": true,
          "change": 2,
          "willSave": false,
          "willSaveWaitUntil": false,
          "save": {
            "includeText": false
          }
        },
        "hoverProvider": true,
        "completionProvider": {
          "resolveProvider": false,
          "triggerCharacters": [
            "$",
            ">",
            "\\",
            ":"
          ]
        },
        "definitionProvider": true,
        "referencesProvider": true,
        "documentHighlightProvider": true,
        "documentSymbolProvider": true,
        "workspaceSymbolProvider": true,
        "codeActionProvider": false,
        "documentFormattingProvider": true,
        "documentRangeFormattingProvider": true,
        "renameProvider": false,
        "typeCoverageProvider": true
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "method": "window/progress",
    "params": {
      "id": 1,
      "label": "Hack: preparing to check edits"
    }
  },
  {
    "jsonrpc": "2.0",
    "method": "window/progress",
    "params": {
      "id": 1,
      "label": "Hack: checking edits"
    }
  },
  {
    "jsonrpc": "2.0",
    "method": "window/actionRequired",
    "params": {
      "id": 2,
      "label": "Hack: save any file to do a whole-program check"
    }
  },
  {
    "jsonrpc": "2.0",
    "method": "window/progress",
    "params": {
      "id": 1,
      "label": null
    }
  },
  {
    "jsonrpc": "2.0",
    "method": "textDocument/publishDiagnostics",
    "params": {
      "uri": "${php_file_uri}",
      "diagnostics": [
        {
          "range": {
            "start": {
              "line": 2,
              "character": 9
            },
            "end": {
              "line": 2,
              "character": 11
            }
          },
          "severity": 1,
          "code": 1002,
          "source": "Hack",
          "message": "Expected identifier",
          "relatedLocations": []
        }
      ]
    }
  }
]
